internal/extension/oop_test.go

```diff
@@ -56,25 +57,62 @@ func TestOOPExtensionManagesExternalProcess(t *testing.T) {
 }
 
 type writerMock struct {
-	messages []string
+	mu            sync.Mutex
+	messages      []string
+	processExited chan struct{}
+	exitOnce      sync.Once
+}
+
+func newWriterMock() *writerMock {
+	return &writerMock{
+		processExited: make(chan struct{}),
+	}
 }
 
 func (w *writerMock) Write(p []byte) (n int, err error) {
-	w.messages = append(w.messages, string(p))
+	w.mu.Lock()
+	defer w.mu.Unlock()
+
+	message := string(p)
+	w.messages = append(w.messages, message)
+
+	// process has completed
+	if strings.Contains(message, "finished with an error") {
+		w.exitOnce.Do(func() {
+			close(w.processExited)
+		})
+	}
+
 	return len(p), nil
 }
 
+func (w *writerMock) getMessages() []string {
+	w.mu.Lock()
+	defer w.mu.Unlock()
+	result := make([]string, len(w.messages))
+	copy(result, w.messages)
+	return result
+}
+
+func (w *writerMock) waitForProcessExit() {
+	<-w.processExited
+}
+
 func TestOOPExtensionForwardsLog(t *testing.T) {
-	writer := &writerMock{}
+	writer := newWriterMock()
 
 	logger := funcr.New(func(_, args string) {
 		writer.Write([]byte(args))
 	}, funcr.Options{})
 
+	socketPath := fmt.Sprintf("/tmp/kuadrant-test-oop-%d.sock", os.Getpid())
+	os.Remove(socketPath)
+	defer os.Remove(socketPath)
```

Comment:
Is this intentional to remove and also defer remove? 🤔

Reply:
Oops yeah I think I was doing this because it hadn't been cleaned up from a previous experiment but the defer should do